lib/CL/pocl_mem_management.c

```diff
@@ -1105,3 +1105,26 @@ pocl_cpu_get_ptr (struct pocl_argument *arg, unsigned global_mem_id)
   ptr += arg->offset;
   return (void *)ptr;
 }
+
+void
+pocl_reset_indirect_ptrs (cl_kernel kernel, void **ptrs, size_t n)
+{
+  if (kernel->indirect_raw_ptrs != NULL)
+    {
+      struct _pocl_ptr_list_node *n, *tmp;
+      DL_FOREACH_SAFE (kernel->indirect_raw_ptrs, n, tmp)
+        {
+          free (n);
+        }
+      kernel->indirect_raw_ptrs = NULL;
+    }
+
+  for (size_t i = 0; i < n; ++i)
+    {
+      struct _pocl_ptr_list_node *node
+        = malloc (sizeof (struct _pocl_ptr_list_node));
+      node->ptr = ptrs[i];
```

Comment:
Should the pointers be checked for NULLs or invalid ones (pointers that are not members of any OpenCL allocations) and filtered out?

Reply:
Yes, we could, but not sure if we _should_, the spec doesn't say it explicitly: https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/clSetKernelExecInfo.html

Reply:
Added a middle-ground solution where I just filter out NULL or unknown pointers for now.